multimodal/src/autogluon/multimodal/learners/default_learner.py

```diff
@@ -0,0 +1,2906 @@
+        #         trainable_param_names=trainable_param_names,
+        #         **metrics_kwargs,
+        #         **optimization_kwargs,
+        #     )
+        # elif self._problem_type == OBJECT_DETECTION:
+        #     task = MMDetLitModule(
+        #         model=model,
+        #         **metrics_kwargs,
+        #         **optimization_kwargs,
+        #     )
+        # else:
+        #     task = LitModule(
+        #         model=model,
+        #         loss_func=loss_func,
+        #         efficient_finetune=OmegaConf.select(config, "optimization.efficient_finetune"),
+        #         mixup_fn=mixup_fn,
+        #         mixup_off_epoch=OmegaConf.select(config, "data.mixup.turn_off_epoch"),
+        #         model_postprocess_fn=model_postprocess_fn,
+        #         trainable_param_names=trainable_param_names,
+        #         skip_final_val=OmegaConf.select(config, "optimization.skip_final_val", default=False),
+        #         **metrics_kwargs,
+        #         **optimization_kwargs,
+        #     )
+
+        task = self._setup_task_lightning_module(
+            trainable_param_names=trainable_param_names,
+            mixup_fn=mixup_fn,
+            loss_func=loss_func,
+            optimization_kwargs=optimization_kwargs,
+            metrics_kwargs=metrics_kwargs,
+        )
+
+        # 20. Setup call backs
+        checkpoint_callback = self._create_checkpoint_callback(minmax_mode=minmax_mode, save_path=save_path, task=task)
+        early_stopping_callback = self._create_early_stop_callback(
+            validation_metric_name=validation_metric_name, minmax_mode=minmax_mode, task=task
+        )
+        lr_callback = self._create_lr_callback()
+        model_summary = pl.callbacks.ModelSummary(max_depth=1)
+        callbacks = [
+            checkpoint_callback,
+            early_stopping_callback,
+            lr_callback,
+            model_summary,
+        ]
+
+        # 21. for hpo with ray
+        # TODO: Leave this out for now. We will add this back later
+        # use_ray_lightning = "_ray_lightning_plugin" in hpo_kwargs
+        # if hpo_mode:
+        #     if use_ray_lightning:
+        #         from ray_lightning.tune import TuneReportCheckpointCallback
+        #     else:
+        #         from ray.tune.integration.pytorch_lightning import TuneReportCheckpointCallback
+        #     tune_report_callback = TuneReportCheckpointCallback(
+        #         {f"{task.validation_metric_name}": f"{task.validation_metric_name}"},
+        #         filename=RAY_TUNE_CHECKPOINT,
+        #     )
+        #     callbacks = [
+        #         tune_report_callback,
+        #         early_stopping_callback,
+        #         lr_callback,
+        #         model_summary,
+        #     ]
+
+        custom_checkpoint_plugin = AutoMMModelCheckpointIO(
+            trainable_param_names=trainable_param_names,
+            model_name_to_id=self._model.name_to_id,
+        )
```

Comment:
Consider using `_get_lightning_plugins()` to wrap this. The function can return a list of plugins.

Reply:
Done